streams/src/test/java/org/apache/kafka/streams/StreamsBuilderTest.java

```diff
@@ -143,15 +143,16 @@ public void shouldAllowJoinUnmaterializedJoinedKTable() {
     public void shouldAllowJoinMaterializedJoinedKTable() {
         final KTable<Bytes, String> table1 = builder.table("table-topic1");
         final KTable<Bytes, String> table2 = builder.table("table-topic2");
-        builder.<Bytes, String>stream("stream-topic").join(table1.join(table2, MockValueJoiner.TOSTRING_JOINER, Materialized.<Bytes, String, KeyValueStore<Bytes, byte[]>>as("store")), MockValueJoiner.TOSTRING_JOINER);
+        KTable<Bytes, String> joinedTable = table1.join(table2, MockValueJoiner.TOSTRING_JOINER, Materialized.<Bytes, String, KeyValueStore<Bytes, byte[]>>as("store"));
+        builder.<Bytes, String>stream("stream-topic").join(joinedTable, MockValueJoiner.TOSTRING_JOINER);
 
         driver.setUp(builder, TestUtils.tempDirectory());
 
         ProcessorTopology topology = builder.internalTopologyBuilder.build();
 
         assertThat(topology.stateStores().size(), equalTo(3));
         assertThat(topology.processorConnectedStateStores("KSTREAM-JOIN-0000000010"), equalTo(Collections.singleton("store")));
-        assertThat(topology.processorConnectedStateStores("KTABLE-MERGE-0000000007"), equalTo(Collections.singleton("store")));
+        assertThat(topology.processorConnectedStateStores("KTABLE-MERGE-0000000006"), equalTo(Collections.singleton("store")));
```

Comment:
This change indicates an backward incompatible change how the build the topology. I think, this must be fixed.

Reply:
I had changed the structure of the test during development.  I have reverted this to the original value, and the test passes.